### PR TITLE
Fix corruption of incoming UTF-8-encoded serial data

### PR DESCRIPTION
**Pull request by palazzol**
### Motivation
Fix for #589

### Change description
Fix corruption of incoming UTF8-encoded serial data

### Other information
Serial data is received in chunks.  When a UTF8-encoded symbol is multibyte, it cannot be decoded properly until all the bytes are available.  If an attempt at decoding is made, it will fail and the symbol for UTF-8 "Replacement Character" � is inserted into the stream.  TextDecoder has a "stream" option to buffer any bytes that cannot be decoded yet, for exactly this reason.  This fix creates an instance of TextDecoder which is persistent for the life of the connection, and then uses that decoder with the stream option to decode UTF8-encoded data properly.

Note - this assumes we expect UTF8 encoded data from the serial port.  Generally speaking, this happens if we have the sketch editor in UTF8 mode, which appears to be the default, and then UTF8 encoded characters are included in a print() statement.

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: arduino-ide-extension/src/node/monitor-service.ts [hunk 1]
```diff
@@ -80,6 +80,7 @@ export class MonitorService extends CoreClientAware implements Disposable {
   private readonly board: Board;
   private readonly port: Port;
   private readonly monitorID: string;
+  private streamingTextDecoder = new TextDecoder('utf8');
```
Comment (z3bra5hax) [inline arduino-ide-extension/src/node/monitor-service.ts:83]:
```suggestion
  private readonly streamingTextDecoder = new TextDecoder('utf8');
```

Comment (z3bra5hax) [reply arduino-ide-extension/src/node/monitor-service.ts:83]: Just to follow convention in reality, no reason this should be modified so let's just update this.

**Review by z3bra5hax — CHANGES_REQUESTED** [review]

File: arduino-ide-extension/src/node/monitor-service.ts (hunk 1 above)
Comment (palazzol) [reply arduino-ide-extension/src/node/monitor-service.ts:83]: Makes sense to me, thanks

**Review by z3bra5hax — APPROVED** [review]

**Review by kittaakos — APPROVED** [review]
This is a quality contribution.

The code looks great and #589 works with the changes.

Thank you!

**Review by per1234 — APPROVED** [review]
I verified this fixes https://github.com/arduino/arduino-ide/issues/589

Thanks @palazzol!

**Pull request merged**